store/snapshots/store.go

```diff
@@ -170,16 +171,21 @@ func (s *Store) Load(height uint64, format uint32) (*types.Snapshot, <-chan io.R
 			chunk, err := s.loadChunkFile(height, format, i)
 			if err != nil {
 				_ = pw.CloseWithError(err)
+				_ = chunk.Close()
 				return
 			}
-			defer chunk.Close()
-			_, err = io.Copy(pw, chunk)
+			err = func() error {
+				defer chunk.Close()
+				if _, err := io.Copy(pw, chunk); err != nil {
+					return fmt.Errorf("failed to copy chunk %d: %w", i, err)
+				}
+				_ = pw.Close()
+				return nil
+			}()
 			if err != nil {
 				_ = pw.CloseWithError(err)
```

Comment:
Sorry for perhaps not being fully clear but also hoist this code into that closure so that it maintains as it was for pw.CloseWithError(err) as that's what I meant by "pw.Close*", basically putting in everything that was in the control flow, into that closure.

Reply:
Hi, I am not quite sure what need to be hoisted, it seems the err value passed to `pw.CloseWithError(err)` is still remained the same in my code

Reply:
After io.Copy, please keep as the code was before, invoking immediately pw.CloseWithError(err). I know right now it functionally is the same but it causes an unnecessary reasoning yet you could have really kept all the code together. Just hoisting it all into that closure and returning an error that you later on just simply check if != nil and return.